agent/stats/engine_integ_test.go

```diff
@@ -220,278 +161,323 @@ func TestStatsEngineWithNewContainers(t *testing.T) {
 			DockerID: container.ID,
 		},
 	})
-	if err != nil {
-		t.Errorf("Failed to write to container change event stream, err: %v", err)
-	}
+	assert.NoError(t, err, "failed to write to container change event stream")
 
 	// Wait for the stats collection go routine to start.
 	time.Sleep(checkPointSleep)
+	validateInstanceMetrics(t, engine)
+	validateEmptyTaskHealthMetrics(t, engine)
 
-	metadata, taskMetrics, err := engine.GetInstanceMetrics()
-	if err != nil {
-		t.Errorf("Error gettting instance metrics: %v", err)
-	}
+	err = client.StopContainer(container.ID, defaultDockerTimeoutSeconds)
+	assert.NoError(t, err, "stopping container failed")
+	// Write the container change event to event stream
+	err = engine.containerChangeEventStream.WriteToEventStream(ecsengine.DockerContainerChangeEvent{
+		Status: api.ContainerStopped,
+		DockerContainerMetadata: ecsengine.DockerContainerMetadata{
+			DockerID: container.ID,
+		},
+	})
+	assert.NoError(t, err, "failed to write to container change event stream")
 
-	err = validateMetricsMetadata(metadata)
-	if err != nil {
-		t.Errorf("Error validating metadata: %v", err)
-	}
+	time.Sleep(waitForCleanupSleep)
 
-	if len(taskMetrics) != 1 {
-		t.Fatalf("Incorrect number of tasks. Expected: 1, got: %d", len(taskMetrics))
-	}
-	taskMetric := taskMetrics[0]
-	if *taskMetric.TaskDefinitionFamily != taskDefinitionFamily {
-		t.Errorf("Excpected task definition family to be: %s, got: %s", taskDefinitionFamily, *taskMetric.TaskDefinitionFamily)
-	}
-	if *taskMetric.TaskDefinitionVersion != taskDefinitionVersion {
-		t.Errorf("Excpected task definition family to be: %s, got: %s", taskDefinitionVersion, *taskMetric.TaskDefinitionVersion)
-	}
+	// Should not contain any metrics after cleanup.
+	validateIdleContainerMetrics(t, engine)
+	validateEmptyTaskHealthMetrics(t, engine)
+}
 
-	err = validateContainerMetrics(taskMetric.ContainerMetrics, 1)
-	if err != nil {
-		t.Errorf("Error validating container metrics: %v", err)
-	}
+func TestStatsEngineWithExistingContainers(t *testing.T) {
+	// Create a new docker stats engine
+	engine := NewDockerStatsEngine(&cfg, dockerClient, eventStream("TestStatsEngineWithExistingContainers"))
+
+	// Create a container to get the container id.
+	container, err := createHealthContainer(client)
+	require.NoError(t, err, "creating container failed")
+	defer client.RemoveContainer(docker.RemoveContainerOptions{
+		ID:    container.ID,
+		Force: true,
+	})
+
+	// Wait for containers from previous tests to transition states.
+	time.Sleep(checkPointSleep)
```

Comment:
I had a question regarding this here as well: https://github.com/aws/amazon-ecs-agent/pull/1186/#discussion_r162112427

Can you please answer why we need to wait for containers from previous tests to transition states at all? Why aren't the tests cleaning up things properly after themselves?